### PR TITLE
mutable flag for txt livedns_records

### PR DESCRIPTION
**Pull request by adriamanu**
The purpose of this PR is to add a **mutable** flag on **livedns_records** to be able to manipulate records that are susceptible of being already existant / modified aside Terraform.
With this pull request, gandi provider will only touch records managed by Terraform by doing **PUT** crud operation from gandi api in order to **update/edit** the list of records.
[This is the related issue](https://github.com/go-gandi/terraform-provider-gandi/issues/106)

### CODE REVIEW

File: gandi/resource_livedns_record.go [hunk 1]
```diff
@@ -74,29 +81,80 @@ func expandRecordID(id string) (zone, name, recordType string, err error) {
...
+		}
+	}
+	return uniqueRecords
+}
+
+func isRecordWrappedWithQuotes(record string) bool {
+	return strings.HasPrefix(record, "\"") && strings.HasSuffix(record, "\"")
+}
+
+func containsRecord(recordsList []string, recordToFind string) int {
```
Comment (nlewo) [inline gandi/resource_livedns_record.go:100]: Could you return a tuple such as `(index int, exist bool)` instead of encoding a non existing record with `-1`?

Comment (nlewo) [inline gandi/resource_livedns_record.go:100]: Could you return a tuple such as `(index int, exist bool)` instead of encoding a non existing record with `-1`?

File: gandi/resource_livedns_record.go [hunk 2]
```diff
@@ -156,13 +241,46 @@ func resourceLiveDNSRecordUpdate(d *schema.ResourceData, meta interface{}) error
...
+		// It means that all resources are managed by Terraform and then records can be safely deleted
+		// Otherwise we need to remove terraform managed records from the records list and update it
+		if len(rec.RrsetValues) == len(valuesList) {
+			if err = client.DeleteDomainRecord(zone, name, recordType); err != nil {
+				return err
+			}
+		} else {
+			var values []string = rec.RrsetValues
+			for _, v := range valuesList {
+				index := containsRecord(values, "\""+v.(string)+"\"")
```
Comment (nlewo) [inline gandi/resource_livedns_record.go:270]:
```suggestion
				index := containsRecord(values, "\""+v.(string)+"\"")
```
```suggestion
				indexm exists := containsRecord(values, "\""+v.(string)+"\"")
```

Comment (nlewo) [inline gandi/resource_livedns_record.go:270]:
```suggestion
				index, exists := containsRecord(values, "\""+v.(string)+"\"")
```
```suggestion
				index := containsRecord(values, "\""+v.(string)+"\"")
```

Comment (nlewo) [inline gandi/resource_livedns_record.go:270]:
```suggestion
				index, exists := containsRecord(values, "\""+v.(string)+"\"")
```

**Review by nlewo — CHANGES_REQUESTED** [review]
Thanks for this nice contribution!

When this first batch of reviews is addressed, I will add an acceptance test (unfortunately, you can't currently do it because it relies on domain that has to exist in the Gandi Sandbox).

**Comment by adriamanu** [comment]
Hello @nlewo , i hope you are doing well.
i updated pr according to your requested changes (some logic modification / unit tests / generate documentation).
Apologies for the delayed update.
Took me some time to test it again and check for edge cases.
Have a good day !

**Comment by nlewo** [comment]
I did some manual tests are write an acceptance test in #117.

Everything looks fine :tada: 

@adriamanu Once this PR is merged, i will rebase #117.

**Comment by adriamanu** [comment]
I'm glad to read that !
@nlewo rebase has been done so that there is only 1 commit to merge (typos have been fixed during the rebasing process).

**Comment by nlewo** [comment]
@adriamanu Thank you!

**Comment by adriamanu** [comment]
Thank you for the support, keep up the good work @nlewo  🙏🏼 !
